packages/lego-bricks/src/components/Accordion/index.tsx

```diff
@@ -26,26 +27,35 @@ export const Accordion = ({
   disabled = false,
   defaultOpen = false,
   animated = true,
+  removeChildrenOnClose = false,
   triggerPosition = 'top',
   wrapperClassName,
   triggerComponent: TriggerComponent,
   children,
 }: Props) => {
-  const childrenWrapperRef = useRef<HTMLDivElement>(null);
-
   const [open, setOpen] = useState(defaultOpen);
   const [initialRender, setInitialRender] = useState(true);
   const [containerHeight, setContainerHeight] = useState(0);
+  const [isTransitioning, setTransitioning] = useState(false);
+
+  // useCallback works with dynamically changing children unlike useRef
+  const childrenWrapperRef = useCallback((node: HTMLDivElement) => {
+    if (!node) return;
+    const resizeObserver = new ResizeObserver(() =>
+      setContainerHeight(node.getBoundingClientRect().height ?? 0),
+    );
+    resizeObserver.observe(node);
+  }, []);
 
   const toggleOpen = () => {
-    setContainerHeight(childrenWrapperRef.current?.clientHeight ?? 0);
     if (disabled) return;
     if (initialRender && animated) {
       // setOpen is triggered through useEffect to ensure proper order of actions
       setInitialRender(false);
     } else {
       setOpen(!open);
     }
+    setTransitioning(true);
   };
 
   // Ensures that the open flag is triggered after the initialRender flag is changed
```

Comment:
With this beautiful ResizeRender thingy, have you checked to see if the following useEffect and the initialRender thingys might not be needed any more?

Reply:
I dont not think to check that to be honest. I just figured to not mess to much with the awesome code that was there before. 😉  I will see if I can clean up this component further.